decentraland/unity-renderer
# fix: sdk7 pointer event pointer blocking

Improved pointer blocking setting for pointer-event entities.

Fixes https://github.com/decentraland/sdk/issues/990

--------------------------

### QA TEST INSTRUCTIONS

1. Enter [this playground session](https://playground.decentraland.org/?explorer-branch=fix/sdk7-pointer-event-pointer-blocker&code=ZnVuY3Rpb24gc2V0dXBVaSgpIHsKICAgIFJlYWN0RWNzUmVuZGVyZXIuc2V0VWlSZW5kZXJlcih1aUNvbXBvbmVudCkKfQoKZnVuY3Rpb24gdWlDb21wb25lbnQgKCl7CiAgICBzd2l0Y2ggKHVpTW9kZSkgewogICAgICAgIGNhc2UgJ21vZGVBJzoKICAgICAgICAgICAgcmV0dXJuIG1vZGVBVWkoKQogICAgICAgIGNhc2UgJ21vZGVCJzoKICAgICAgICAgICAgcmV0dXJuIG1vZGVCVWkoKQogICAgfQp9Cgpjb25zdCBGVUxMX1dJRFRIID0gMjAwCmNvbnN0IEZVTExfSEVJR0hUID0gMjI1CmNvbnN0IEZVTExfUE9TX1ggPSAyMDAKY29uc3QgRlVMTF9QT1NfWSA9IDMyMAoKY29uc3QgSUNPTl9XSURUSCA9IDEwMApjb25zdCBJQ09OX0hFSUdIVCA9IDcyCmNvbnN0IElDT05fUE9TX1ggPSAyMDAKY29uc3QgSUNPTl9QT1NfWSA9IDMyMAoKY29uc3QgQlVUVE9OX1dJRFRIID0gNzAKY29uc3QgQlVUVE9OX0hFSUdIVCA9IDMwCgpjb25zdCBDT0wxX1ggPSA5NQpjb25zdCBDT0wzX1ggPSAxMwoKY29uc3QgUk9XMV9ZID0gMTgyIC8vIHVwIDQxCmNvbnN0IFJPVzVfWSA9IDI3CmNvbnN0IFJPVzZfWSA9IDQKCmNvbnN0IEZVTExfQ0FQVElPTl9YID0gMTAKY29uc3QgRlVMTF9DQVBUSU9OX1dJRFRIID0gNjAKY29uc3QgRlVMTF9DQVBUSU9OX0hFSUdIVCA9IDIwCgpjb25zdCBJQ09OX0NBUFRJT05fWCA9IDAKY29uc3QgSUNPTl9DQVBUSU9OX1dJRFRIID0gMzMKY29uc3QgSUNPTl9DQVBUSU9OX0hFSUdIVCA9IDE1Cgp0eXBlIFVpTW9kZSA9ICdtb2RlQSd8J21vZGVCJwoKbGV0IHVpTW9kZTpVaU1vZGUgPSAnbW9kZUEnCgpmdW5jdGlvbiBtb2RlQlVpKCkgewogICAgcmV0dXJuIDxVaUVudGl0eQogICAgICAgIHVpVHJhbnNmb3JtPXt7CiAgICAgICAgICAgIHBvc2l0aW9uVHlwZTogJ2Fic29sdXRlJywKICAgICAgICAgICAgd2lkdGg6IEZVTExfV0lEVEgsCiAgICAgICAgICAgIGhlaWdodDogRlVMTF9IRUlHSFQsCiAgICAgICAgICAgIHBvc2l0aW9uOiB7IHJpZ2h0OiBGVUxMX1BPU19YLCBib3R0b206IEZVTExfUE9TX1kgfSwKICAgICAgICB9fQogICAgICAgIHVpQmFja2dyb3VuZD17eyBjb2xvcjogQ29sb3I0LmZyb21IZXhTdHJpbmcoIiM3MGFjNzZmZiIpIH19CiAgICA%2BCiAgICAgICAgPEJ1dHRvbiAKICAgICAgICAgICAgdmFsdWU9J0J1dHRvbiAyJwogICAgICAgICAgICBvbk1vdXNlRG93bj17KCk9PnsgCiAgICAgICAgICAgICAgICBjb25zb2xlLmxvZygiQnV0dG9uIDIgUHJlc3NlZCIpCiAgICAgICAgICAgIH19CiAgICAgICAgICAgIHVpVHJhbnNmb3JtPXt7CiAgICAgICAgICAgICAgICBwb3NpdGlvblR5cGU6ICdhYnNvbHV0ZScsCiAgICAgICAgICAgICAgICB3aWR0aDogQlVUVE9OX1dJRFRILAogICAgICAgICAgICAgICAgaGVpZ2h0OiBCVVRUT05fSEVJR0hULAogICAgICAgICAgICAgICAgcG9zaXRpb246IHsgcmlnaHQ6IENPTDFfWCwgYm90dG9tOiBST1cxX1kgfSwKICAgICAgICAgICAgfX0KICAgICAgICA%2BPC9CdXR0b24%2BICAgICAgICAKICAgICAgICA8QnV0dG9uIAogICAgICAgICAgICB2YWx1ZT0nVG9nZ2xlVWknCiAgICAgICAgICAgIG9uTW91c2VEb3duPXsoKT0%2BeyAKICAgICAgICAgICAgICAgIHVpTW9kZSA9ICdtb2RlQSc7IAogICAgICAgICAgICB9fQogICAgICAgICAgICB1aVRyYW5zZm9ybT17ewogICAgICAgICAgICAgICAgcG9zaXRpb25UeXBlOiAnYWJzb2x1dGUnLAogICAgICAgICAgICAgICAgd2lkdGg6IEJVVFRPTl9XSURUSCwKICAgICAgICAgICAgICAgIGhlaWdodDogQlVUVE9OX0hFSUdIVCwKICAgICAgICAgICAgICAgIHBvc2l0aW9uOiB7IHJpZ2h0OiBDT0wzX1gsIGJvdHRvbTogUk9XNV9ZIH0KICAgICAgICAgICAgfX0KICAgICAgICA%2BPC9CdXR0b24%2BCiAgICAgICAgPFVpRW50aXR5IAogICAgICAgICAgICB1aVRyYW5zZm9ybT17eyAKICAgICAgICAgICAgICAgIHBvc2l0aW9uVHlwZTogJ2Fic29sdXRlJywKICAgICAgICAgICAgICAgIHdpZHRoOiBGVUxMX0NBUFRJT05fV0lEVEgsIAogICAgICAgICAgICAgICAgaGVpZ2h0OiBGVUxMX0NBUFRJT05fSEVJR0hULAogICAgICAgICAgICAgICAgcG9zaXRpb246IHsgcmlnaHQ6IEZVTExfQ0FQVElPTl9YLCBib3R0b206IFJPVzZfWSB9IAogICAgICAgICAgICB9fSA%2BCiAgICAgICAgICAgIDxMYWJlbAogICAgICAgICAgICAgICAgdmFsdWU9IkV4cGFuZGVkIFVJIgogICAgICAgICAgICAgICAgY29sb3I9e0NvbG9yNC5XaGl0ZSgpfQogICAgICAgICAgICAgICAgZm9udFNpemU9ezE0fQogICAgICAgICAgICAgICAgZm9udD0ic2Fucy1zZXJpZiIKICAgICAgICAgICAgICAgIHRleHRBbGlnbj0ibWlkZGxlLWNlbnRlciIKICAgICAgICAgICAgLz4KICAgICAgICA8L1VpRW50aXR5PgogICAgPC9VaUVudGl0eT4KfQoKZnVuY3Rpb24gbW9kZUFVaSgpIHsKICAgIHJldHVybiA8VWlFbnRpdHkKICAgICAgICB1aVRyYW5zZm9ybT17ewogICAgICAgICAgICBwb3NpdGlvblR5cGU6ICdhYnNvbHV0ZScsICAgICAgICAKICAgICAgICAgICAgd2lkdGg6IElDT05fV0lEVEgsCiAgICAgICAgICAgIGhlaWdodDogSUNPTl9IRUlHSFQsCiAgICAgICAgICAgIHBvc2l0aW9uOiB7IHJpZ2h0OiBJQ09OX1BPU19YLCBib3R0b206IElDT05fUE9TX1kgfSwgCiAgICAgICAgfX0KICAgICAgICB1aUJhY2tncm91bmQ9e3sgY29sb3I6IENvbG9yNC5mcm9tSGV4U3RyaW5nKCIjNzBhYzc2ZmYiKSB9fQogICAgPgogICAgICAgIDxCdXR0b24gCiAgICAgICAgICAgIHZhbHVlPSdUb2dnbGUgVUknIC8vIHVzaW5nIHRleHQgb25seSBhcyBkb2N1bWVudGF0aW9uIChidXR0b24gbGFiZWwpaGVyZS4KICAgICAgICAgICAgb25Nb3VzZURvd249eygpPT57IAogICAgICAgICAgICAgICAgdWlNb2RlID0gJ21vZGVCJzsKICAgICAgICAgICAgfX0KICAgICAgICAgICAgY29sb3I9e0NvbG9yNC5CbGFjaygpfS8vZnJvbUhleFN0cmluZygiI0ZGRkZGRjAwIil9IC8vIHRvIGhpZGUgdGhlIHRleHQKICAgICAgICAgICAgdWlUcmFuc2Zvcm09e3sKICAgICAgICAgICAgICAgIHBvc2l0aW9uVHlwZTogJ2Fic29sdXRlJywKICAgICAgICAgICAgICAgIHdpZHRoOiBCVVRUT05fV0lEVEgsCiAgICAgICAgICAgICAgICBoZWlnaHQ6IEJVVFRPTl9IRUlHSFQsCiAgICAgICAgICAgICAgICBwb3NpdGlvbjogeyByaWdodDogQ09MM19YLCBib3R0b206IFJPVzVfWSB9LAogICAgICAgICAgICB9fQogICAgICAgID48L0J1dHRvbj4KICAgICAgICA8VWlFbnRpdHkgCiAgICAgICAgICAgIHVpVHJhbnNmb3JtPXt7IAogICAgICAgICAgICAgICAgcG9zaXRpb25UeXBlOiAnYWJzb2x1dGUnLAogICAgICAgICAgICAgICAgd2lkdGg6IElDT05fQ0FQVElPTl9XSURUSCwgCiAgICAgICAgICAgICAgICBoZWlnaHQ6IElDT05fQ0FQVElPTl9IRUlHSFQsCiAgICAgICAgICAgICAgICBwb3NpdGlvbjogeyByaWdodDogSUNPTl9DQVBUSU9OX1gsIGJvdHRvbTogUk9XNl9ZIH0gCiAgICAgICAgICAgIH19ID4KICAgICAgICAgICAgPExhYmVsCiAgICAgICAgICAgICAgICB2YWx1ZT0iU21hbGwgVUkiCiAgICAgICAgICAgICAgICBjb2xvcj17Q29sb3I0LldoaXRlKCl9CiAgICAgICAgICAgICAgICBmb250U2l6ZT17MTB9CiAgICAgICAgICAgICAgICBmb250PSJzYW5zLXNlcmlmIgogICAgICAgICAgICAgICAgdGV4dEFsaWduPSJtaWRkbGUtY2VudGVyIgogICAgICAgICAgICAvPgogICAgICAgIDwvVWlFbnRpdHk%2BICAgICAgICAKICAgIDwvVWlFbnRpdHk%2BCn0KCnNldHVwVWkoKQ%3D%3D)
2. After the world loads hit the RUN button
3. Confirm that every red button in the UI is clickable (instead of letting the pointer/mouse trespass it and getting locked)
